### PR TITLE
Added BIDS utilities to handle BIDS-compliant datasets

### PR DESCRIPTION
**Pull request by estefanysuarez**
### Add BIDS utilities for EEG and iEEG data management
This PR introduces a new bids_utils module that provides utilities for working with Brain Imaging Data Structure (BIDS) compliant neuroimaging data, particularly for EEG and iEEG recordings. 

Key Features:
- Discover EEG and iEEG recordings from BIDS directories or a list of candidate files
- Build MNE-BIDS paths for seamless integration with the MNE ecosystem
- Check if data files matching a recording exist locally
- Support for multiple BIDS-compliant formats (.edf, .vhdr, .set, .bdf, .nwb)
- Testing: Comprehensive test coverage for all functions including edge cases and error conditions.

### Extended functionality for `extract_channels` in mne_utils
- The new function works for both EEG and iEEG MNE Raw objects. If `channels_name_mapping` or `channels_type_mapping` dictioneries are passed, then channel names and types will be remapped. This can replace `_build_channels` in OpenNeuroPipeline classes. 
- A new 'status' flag is added to identify bad channels
- Coordinates will be extracted if the exist in the MNE Raw object
- Tests were updated to include the new features

**Note:** Since Open Neuro datasets are BIDS compliant and MNE compatible, these utilities can be reused for the OpenNeuro Pipelines ✨

### CODE REVIEW

File: brainsets/utils/bids_utils.py [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+        fname: The BIDS filename to parse (e.g., 'sub-01_task-Sleep_eeg.edf') or a Path object
+        modality: The anticipated modality suffix (e.g., 'eeg', 'ieeg'). If set to None, filenames will be parsed regardless of their suffix.
+
+    Returns:
+        Dictionary with keys: subject, session, task, acquisition, run, description
+        All values are prefix-free (e.g., subject='01', not 'sub-01')
+        Returns None if the filename doesn't match BIDS pattern or modality
+    """
+    try:
+        bids_path = get_bids_path_from_fname(fname, check=False)
```
Comment (milosobral) [inline brainsets/utils/bids_utils.py]: Is there a reason why you turn the check off here? Is it because we assume that anything that gets passed to this function is MNE compliant?

File: brainsets/utils/mne_utils.py [hunk 2]
```diff
@@ -80,24 +82,107 @@ def extract_eeg_signal(
 
 
 def extract_channels(
-    recording_data: "mne.io.Raw",
+    recording_data: mne.io.Raw,
+    channels_name_mapping: dict[str, str] | None = None,
+    channels_type_mapping: dict[str, list[str]] | None = None,
```
Comment (milosobral) [inline brainsets/utils/mne_utils.py]: For this function, it would maybe be better to have these two arguments be functions. Users can call this `extract_channels` function and pass any transformation function that they want to be applied to the channels.id and channels.type. That way, we make less assumptions about what people might want to do with their channels and how they want to apply it

File: brainsets/utils/mne_utils.py [hunk 3]
```diff
@@ -80,24 +82,107 @@ def extract_eeg_signal(
...
+        "type": channel_types,
+        "status": status,
+    }
+    if not np.all(np.isnan(x_coords)):
+        channel_fields["x"] = x_coords
+    if not np.all(np.isnan(y_coords)):
+        channel_fields["y"] = y_coords
+    if not np.all(np.isnan(z_coords)):
+        channel_fields["z"] = z_coords
+
```
[the remark this answers is not in the document]
Comment (estefanysuarez) [reply brainsets/utils/mne_utils.py]: Makes sense. I can do that ;)

File: brainsets/utils/bids_utils.py [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+
+def get_subject_info(
+    subject_id: str,
+    participants_data: Optional[pd.DataFrame] = None,
+    bids_root: Optional[Path | str] = None,
+) -> dict:
+    """Return subject information dict with 'age' and 'sex' keys.
+
+    Override this method to provide subject information from alternative sources
+    (e.g., custom metadata files, databases, etc.).
```
[the remark this answers is not in the document]
Comment (estefanysuarez) [reply brainsets/utils/bids_utils.py]: Done
